sycl/doc/extensions/supported/sycl_ext_intel_esimd/examples/histogram_2d.md

```diff
@@ -0,0 +1,186 @@
+
+    Compile and run :
+```bash > clang++ - fsycl histogram_2d.cpp
+> ONEAPI_DEVICE_SELECTOR = level_zero:gpu./ a.out 
+Running on Intel(R) UHD Graphics 630 
+Processing inputs
+GPU Histogram :
+.......................
+CPU Histogram : 
+........................ 
+Passed
+```
+
+Source code :
+```C++
+#include <array>
+#include <iostream>
+#include <sycl/ext/intel/esimd.hpp>
+#include <sycl/sycl.hpp>
+
+using namespace sycl;
+
+#define NUM_BINS 256
+#define IMG_WIDTH 1024
+#define IMG_HEIGHT 1024
+//
+// each parallel_for handles 1x32 bytes
+//
+#define BLOCK_WIDTH 32
+#define BLOCK_HEIGHT 1
+
+void histogram_CPU(unsigned int width, unsigned int height, unsigned char *srcY,
+                   unsigned int *cpuHistogram) {
+  int i;
+  for (i = 0; i < width * height; i++) {
+    cpuHistogram[srcY[i]] += 1;
+  }
+}
+
+void writeHist(unsigned int *hist) {
+  int total = 0;
+
+  std::cerr << " Histogram: \n";
+  for (int i = 0; i < NUM_BINS; i += 8) {
+    std::cerr << "\n  [" << i << " - " << i + 7 << "]:";
+    for (int j = 0; j < 8; j++) {
+      std::cerr << "\t" << hist[i + j];
+      total += hist[i + j];
+    }
+  }
+  std::cerr << "\nTotal = " << total << " \n";
+}
+
+int checkHistogram(unsigned int *refHistogram, unsigned int *hist) {
+
+  for (int i = 0; i < NUM_BINS; i++) {
+    if (refHistogram[i] != hist[i]) {
+      return 0;
+    }
+  }
+  return 1;
+}
+
+int main(int argc, char *argv[]) {
+  unsigned int width = IMG_WIDTH * sizeof(unsigned int);
+  unsigned int height = IMG_HEIGHT;
+
+  // ------------------------------------------------------------------------
+
+  // Allocate Input Buffer
+  queue q;
+
+  auto dev = q.get_device();
+  unsigned char *srcY = malloc_shared<unsigned char>(width * height, q);
+  if (srcY == NULL) {
+    std::cerr << "Out of memory\n";
+    exit(1);
+  }
+
+  unsigned int *bins = malloc_shared<unsigned int>(NUM_BINS, q);
+  std::cout << "Running on " << dev.get_info<info::device::name>() << "\n";
+
+  uint range_width = width / BLOCK_WIDTH;
+  uint range_height = height / BLOCK_HEIGHT;
+
+  // Initializes input.
+  unsigned int input_size = width * height;
+  std::cerr << "Processing inputs\n";
+
+  srand(2009);
+  for (int i = 0; i < input_size; ++i) {
+    srcY[i] = rand() % 256;
+  }
+
+  for (int i = 0; i < NUM_BINS; i++) {
+    bins[i] = 0;
+  }
+
+  // ------------------------------------------------------------------------
+  // CPU Execution:
+
+  unsigned int cpuHistogram[NUM_BINS];
+  memset(cpuHistogram, 0, sizeof(cpuHistogram));
+  histogram_CPU(width, height, srcY, cpuHistogram);
+
+  try {
+    // create ranges
+    // We need that many workitems
+    auto GlobalRange = range<2>(range_width, range_height);
+    // Number of workitems in a workgroup
+    auto LocalRange = range<2>(1, 1);
+    nd_range<2> Range(GlobalRange, LocalRange);
+
+    auto e = q.submit([&](handler &cgh) {
+      cgh.parallel_for<class Hist>(
+          Range, [=](nd_item<2> ndi) SYCL_ESIMD_KERNEL {
```

Comment:
Please minimize the code by removing all not strictly required for better understanding.
Making the code more compact (here and in all other places) helps to find the right place and understand the idea faster.
```suggestion
    q.parallel_for(Range, [=](nd_item<2> ndi) SYCL_ESIMD_KERNEL {
```